packages/react-app/src/App.jsx

```diff
@@ -10,32 +10,75 @@ import IconButton from '@mui/material/IconButton'
 import Toast from 'components/Toast'
 import { BadgeContext } from 'contexts/BadgeContext'
 import externalContracts from 'contracts/external_contracts'
-import { switchToOptimism } from 'helpers/SwitchToOptimism'
+import { useUserProviderAndSigner } from 'eth-hooks'
 const { ethers } = require('ethers')
 
-function App({ mainnet, localProvider, appChainId }) {
+const APPSTATEACTION = {
+  GOERLICHAINID: '5',
+  OPTIMISMCHAINID: '10',
+}
+
+const defaultState = {
+  provider: new ethers.providers.Web3Provider(window.ethereum),
+  chainId: '5',
+  contractRef: externalContracts['5'].contracts.REMIX_REWARD,
+}
+
+function appStateReducer(state, actionType) {
+  switch (actionType.type) {
+    case '10':
+      const optimism = {
+        provider: new ethers.providers.Web3Provider(window.ethereum),
+        chainid: '10',
+        contractRef: externalContracts['10'].contracts.REMIX_REWARD,
+      }
+      return optimism
+    case '5':
+      const goerli = {
+        provider: new ethers.providers.Web3Provider(window.ethereum),
+        chainid: '1',
+        contractRef: externalContracts['1'].contracts.REMIX_REWARD,
```

Comment:
this should be `externalContracts['5'].contracts.REMIX_REWARD,`

Reply:
I have fixed this mismatch also